plugins/salesforce/core/src/main/java/org/pentaho/di/trans/steps/salesforceinput/SalesforceInput.java

```
@@ -438,4 +460,189 @@ public void dispose( StepMetaInterface smi, StepDataInterface sdi ) {
+      response.put( StepInterface.STATUS, StepInterface.FAILURE_STATUS );
+    }
+    return response;
+  }
+
+  public FieldsResponse getFields() throws Exception {
+    SalesforceConnection connection = null;
+    try {
+      SalesforceInputMeta meta = (SalesforceInputMeta) getStepMetaInterface();
+      String realURL = getTransMeta().environmentSubstitute( meta.getTargetURL() );
+      String realUsername = getTransMeta().environmentSubstitute( meta.getUsername() );
+      String realPassword = Utils.resolvePassword( getTransMeta(), meta.getPassword() );
+      int realTimeOut = Const.toInt( getTransMeta().environmentSubstitute( meta.getTimeout() ), 0 );
+
+      connection = new SalesforceConnection( log, realURL, realUsername, realPassword );
+      connection.setTimeOut( realTimeOut );
+      FieldsResponse fieldsResponse = new FieldsResponse( new ArrayList<>(), new HashSet<>() );
+      if ( meta.isSpecifyQuery() ) {
+        // Free hand SOQL
+        String realQuery = getTransMeta().environmentSubstitute( meta.getQuery() );
+        connection.setSQL( realQuery );
+        connection.connect();
+        // We are connected, so let's query
+        XmlObject[] fields = connection.getElements();
+        int nrFields = fields.length;
+        for ( int i = 0; i < nrFields; i++ ) {
+          fieldsResponse.fieldDTOList.addAll( addFields( "", fieldsResponse.fieldNames, fields[ i ] ) );
+        }
+      } else {
+        connection.connect();
+
+        Field[] fields = connection.getObjectFields( meta.getModule() );
+        for ( int i = 0; i < fields.length; i++ ) {
+          Field field = fields[ i ];
+          fieldsResponse.fieldNames.add( field.getName() );
+          fieldsResponse.fieldDTOList.add( addField( field ) );
+        }
+      }
+      return fieldsResponse;
+
+    } finally {
+      if ( connection != null ) {
+        try {
+          connection.close();
+        } catch ( Exception e ) {
+          // Ignore errors
```

why are you ignoring the error here in  SalesforceInput.java#addFields? 

Shouldn't the exception be at least logged as trace or debug?

---

Addressed